form/form.go

```diff
@@ -414,7 +414,14 @@ func isBool(f *descriptor.FieldDescriptorProto) bool {
 }
 
 func isString(f *descriptor.FieldDescriptorProto) bool {
-	return f.GetType() == descriptor.FieldDescriptorProto_TYPE_STRING
+	switch f.GetType() {
```

Comment:
So isString is returning true for Int64, etc.
Is this what we want?

Reply:
Could we use Number.isSafeInteger or somehow keep the smaller integers that were working, working the way they were.
And only handle integers as strings when they are too large?

I ask this because a project that already has a link that they shared with someone, will probably stop working once it requires int64s to be strings or not?

Reply:
Do you mean it will break the compatibility ?

Your consideration is meaningful as for compatibility, but the `isSafeInteger` is really hack, right ? (XD

I mean I will consider to work on this .

Reply:
Hi, @awalterschulze  sry for missing these days.

I review the again, and I find that the `isString` is defined without using it. 
As for your consideration: 
* I fixed the `isNumber`(it can be used as isSafeInteger now),which means it can now deal with the correct Number type.  
* We do not know whether the type int64's real number is over int32 or not ,so, I think we can not do that in generating the form template.
* How about  do this in "URL decoding"(I mean to keep the compatibility, we can convert the `number` to `string` in server-side when the defined type is int64 but the GET method pass the int32 to the server-side)?

Reply:
If isString is not being called, can we change it back?

Reply:
We need to consider that someone has probably shared a url with a number (in the safe range), that will be used to populate the form.  So we need to try decoding it both ways, as a number for the safe range and as a string in the non safe range.  This way we can also url encode it as a number in the safe range and as a string in the non safe range.   What do you think?

Reply:
> If isString is not being called, can we change it back?

I think we can't. INT64 in proto3 should be treated as STRING, right?

Reply:
> We need to consider that someone has probably shared a url with a number (in the safe range), that will be used to populate the form. So we need to try decoding it both ways, as a number for the safe range and as a string in the non safe range. This way we can also url encode it as a number in the safe range and as a string in the non safe range. What do you think?

I think I did this in my latest commit.

Reply:
Yes I can now finally see, that you are checking for string, but still allow normal int.

But I still can't decide about this `isString` function.
But maybe first address my other comments and then I'll check this diff out and try it for myself.